### ISSUE TITLE
Enable tables which interrupt a paragraph (like GFM does)

### ISSUE THREAD

**Issue by minaminao** [miyuchina/mistletoe#166]
Input markdown:
```
A:
|a|b|
|-|-|
|a|b|
```

`ast_renderer` outputs:
```
    {
      "type": "Paragraph",
      "children": [
        {
          "type": "RawText",
          "content": "A:"
        },
        {
          "type": "LineBreak",
          "soft": true,
          "content": ""
        },
        {
          "type": "RawText",
          "content": "|a|b|"
        },
        {
          "type": "LineBreak",
          "soft": true,
          "content": ""
        },
        {
          "type": "RawText",
          "content": "|-|-|"
        },
        {
          "type": "LineBreak",
          "soft": true,
          "content": ""
        },
        {
          "type": "RawText",
          "content": "|a|b|"
        }
      ]
    },
```

I believe `Table` should be output, not `RawText`.

#### Actual output in GitHub:
A:
|a|b|
|-|-|
|a|b|

**Comment by pbodnar** [comment]
@minaminao, thanks for the report. After some analysis, I would consider this as a change request which could be stated like this:

_Enable tables to interrupt a paragraph_

Let me explain why:

1. Table like this works fine:

**Comment by pbodnar** [comment]
FYI, by mistake, I have submitted the previous comment before finishing it. It is ready now. :)

**Comment by minaminao** [comment]
Thanks for the analysis :)

(I found this in a table with more than 3 consecutive `-`'s, but simplified the table for reporting, and then have related it to another isssue #131 unexpectedly)

**Comment by anderskaplan** [comment]
As mistletoe has the ambition to support custom tokens, it would be nice if it also had a mechanism to customize which tokens are allowed to interrupt a paragraph. As you point out @pbodnar , this is currently hard coded in `Paragraph.read()`. With such a mechanism in place, it should be straightforward to enable the desired behavior for tables.

I have some ideas about how to do this, so I'll try and make a PR out of them.

**Comment by pbodnar** [comment]
@anderskaplan, that looks promising - having it implemented in a more generic way, not just for `Table`, would be great. :)

**Comment by anderskaplan** [comment]
Created two PR's: one "infrastructure" fix (#186) and a draft PR for the new mechanism + table change (#187).
